### DISCUSSION TITLE
support of Raspberry Pi camera model 3

### DISCUSSION THREAD

**Discussion by eravion** [Motion-Project/motion#1626] [Q&A] [answered]
Hello,

Could it be possible to add support to Raspberry Pi camera 3?

It is not visible on the list as not bcm2835 compatible.
It is seen as unicam.
$ v4l2-ctl  --list-devices
unicam (platform:3f801000.csi):
	/dev/video0
	/dev/video1
	/dev/media3

bcm2835-codec-decode (platform:bcm2835-codec):
	/dev/video10
	/dev/video11
	/dev/video12
	/dev/video18
	/dev/video31
	/dev/media0

bcm2835-isp (platform:bcm2835-isp):
	/dev/video13
	/dev/video14
	/dev/video15
	/dev/video16
	/dev/video20
	/dev/video21
	/dev/video22
	/dev/video23
	/dev/media1
	/dev/media2

$ dmesg | grep imx
[   10.926280] imx708 10-001a: camera module ID 0x0301
[   10.930276] imx708 10-001a: Consider updating driver imx708 to match on endpoints

Extract of motion.log

[1:ml1] [ERR] [VID] [Jan 28 17:40:06] v4l2_fps_set: Error setting fps. Return code -1
[1:ml1] [ERR] [VID] [Jan 28 17:40:06] v4l2_mmap_set: Error starting stream. VIDIOC_STREAMON: Invalid argument
[1:ml1] [ERR] [VID] [Jan 28 17:40:06] vid_start: V4L2 device failed to open

**Comment by Mr-Dave** [comment, marked as answer]
Yea....so this confused me a lot too under the new raspian versions...

They got rid of being able to just use tools directly against that appears to be v4l2 devices and instead now require that it be prefixed with a `libcamerify  v4l2-ctl --list-devices`  So for motion to work, you have to start it as `libcamerify motion`

**Reply by eravion** [reply]
Perfect it works.
Thanks a lot!
